### PR TITLE
Support systemd service enablement (bnc#908118)

### PR DESCRIPTION
**Pull request by tserong**
Signed-off-by: Tim Serong <tserong@suse.com>

### CODE REVIEW

File: chef/cookbooks/ceph/recipes/osd.rb [hunk 1]
```diff
@@ -192,8 +195,12 @@
         action [ :enable, :start ]
         supports :restart => true
         subscribes :restart, resources(:template => "/etc/ceph/ceph.conf")
-      end
+      end if service_type != "systemd"
 
+      # In addition to the osd services, ceph.target must be enabled when using systemd
+      service "ceph.target" do
+        action :enable
+      end if service_type == "systemd"
```
Comment (tboerger) [inline chef/cookbooks/ceph/recipes/osd.rb]: ```service_type``` is IMO the best way to detect that :)

**Comment by tboerger** [comment]
As we can use the ``ìnit_style```information on various places i would prefer to backport https://github.com/opscode/ohai/blob/master/lib/ohai/plugins/init_package.rb for ohai 6.

**Comment by vuntz** [comment]
+1

**Comment by dirkmueller** [comment]
+1

**Comment by tboerger** [comment]
Nice, no reaction for other improvements -.-

**Pull request merged**